modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/EntityModelSchemaBasedEdmProvider.java

```diff
@@ -104,8 +108,15 @@ private CsdlComplexType _createCsdlComplexType(
 				if (Objects.equals(
 						curEntityField.getType(), EntityField.Type.COMPLEX)) {
 
-					csdlComplexTypes.addAll(
-						_createCsdlComplexTypes(entityFieldsMap, _NAMESPACE));
+					ComplexEntityField curComplexEntityField =
+						(ComplexEntityField)curEntityField;
+
+					_handleComplexType(
+						complexTypeNameDepth,
+						_createCsdlComplexType(
+							complexTypeNameDepth, curComplexEntityField,
+							_NAMESPACE, depth++),
```

Comment:
the depth + 1 value is used in the following line? Should it be included in the map with the depth + 1 or the current depth?

Reply:
I don't know if it is correct, but I think it would be more clear just passing `depth + 1`